.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,64 @@
+name: Publish on Github container registry
+
+on:
+  release:
+    type: [published]
+  push:
+    branches:
+      - main
+      - feat/ci_docker_build
+  pull_request:
+    paths:
+      - 'pyproject.toml'
+      - 'Dockerfile'
+
+env:
+  REGISTRY: ghcr.io
+
+jobs:
+  build_and_push:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      # https://github.com/actions/checkout
+      - name: checkout repository
+        uses: actions/checkout@v4
+
+      - name: lowercase image name
+        run: |
+          echo "IMAGE_NAME=${GITHUB_REPOSITORY,,}" >> ${GITHUB_ENV}
+
+      - name: Get current release version
+        id: release-version
+        run: |
+          version=$(grep -E '^version += +' pyproject.toml | sed -E 's/.*= +//' | sed "s/['\"]//g")
+          echo "version=${version}" >> $GITHUB_OUTPUT
+
+      # https://github.com/docker/login-action
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3.0.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      # https://github.com/docker/metadata-action
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5.0.0
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: |
+            type=raw,value=latest,enable=${{ github.event_name == 'push' }}
+            type=raw,value=${{ steps.release-version.outputs.version }},enable=${{ github.event_name == 'release' }}
```

Comment:
**question:** what happens to the tag when the workflow is triggered by `pull_request`?

Reply:
> Note: Assuming this would fall into "push" and the latest tag would be added